src/System.Text.Json/docs/ReferenceHandling_spec.md

```diff
@@ -0,0 +1,433 @@
+* **System.Array**
+
+Aside from above, the Deserializer will throw when a CLR value type is passed as preserved when `ReferenceHandling.Preserve` is set. 
+
+As a rule of thumb, we should throw on all cases where the payload contains metadata that is impossible to create with the `JsonSerializer`, however, this conflicts with feature parity in Json.Net; those scenarios are described below.
+
+## Reference objects ($ref)
+
+* Regular property **before** `$ref`.
+  * **Json.Net**: `$ref` is ignored if a regular property is previously found in the object.
+  * **S.T.Json**: Throw - Reference objects cannot contain other properties.
+
+```json
+{
+    "$id": "1",
+    "Name": "Angela",
+    "ReportsTo": {
+        "Name": "Bob",
+        "$ref": "1"
+    }
+}
+```
+
+* Regular property **after** `$ref`.
+  * **Json.Net**: Throw - Additional content found in JSON reference object.
+  * **S.T.Json**: Throw - Reference objects cannot contain other properties.
+
+```json
+{
+    "$id": "1",
+    "Name": "Angela",
+    "ReportsTo":{
+        "$ref": "1",
+        "Name": "Angela" 
+    }
+}
+```
+ 
+* Metadata property **before** `$ref`:
+  * **Json.Net**: `$id` is disregarded and the reference is set.
+  * **S.T.Json**: Throw - Reference objects cannot contain other properties.
+```json
+{
+    "$id": "1",
+    "Name": "Angela",
+    "ReportsTo": {
+        "$id": "2",
+        "$ref": "1"
+    }
+}
+```
+
+* Metadata property **after** `$ref`:
+  * **Json.Net**: Throw with the next message: 'Additional content found in JSON reference object'.
+  * **S.T.Json**: Throw - Reference objects cannot contain other properties.
+```json
+{
+    "$id": "1",
+    "Name": "Angela",
+    "ReportsTo": {
+        "$ref": "1",
+        "$id": "2"
+    }
+}
+```
+ 
+* reference object is before preserved object (or preserved object was never spotted):
+  * **Json.Net**: Reference object evaluates as `null`.
+  * **S.T.Json**: Reference object evaluates as `null`.
+```json
+[
+    {
+        "$ref": "1"
+    },
+    {
+        "$id": "1",
+        "Name": "Angela"
+    }
+]
+```
+
+## Preserved objects ($id)
+
+* Having more than one `$id` in the same object:
+  * **Json.Net**: last one wins, in the example, the reference object evaluates to `null` (if `$ref` would be `"2"`, it would evaluate to itself).
+  * **S.T.Json**: Throw - Object already defines a reference identifier.
+```json
+{
+    "$id": "1",
+    "$id": "2",
+    "Name": "Angela",
+    "ReportsTo": {
+        "$ref": "1"
+    }
+}
+```
+
+* `$id` is not the first property:
+  * **Json.Net**: Object is not preserved and cannot be referenced, therefore any reference to it would evaluate as null.
+
+  * **S.T.Json**: We can handle the `$id` not being the first property since we store the reference at the moment we spot the property, I don't think we should throw but keep in mind that this is not a normal payload produced by the serializer.
```

Comment:
Why are we deviating from our general rule of thumb here? What's the benefit?

Reply:
It might be that something reads our payload and when emitting it back (i.e: JavaScript), the order of the properties change? But then that also might apply to arrays (`$values` being above `$id`). 

I will switch to throw in this case.

Reply:
We can throw for this case according to our general rule of thumb and then add a note that we can theoretically support this with little change to the implementation, if needed.